### DISCUSSION TITLE
A question about multi platform support

### DISCUSSION THREAD

**Discussion by justasxz** [dlemstra/Magick.NET#1716] [Help]
Hi, I am developing an application, that uses Magick.Net and is supposed to work with blazor hybrid. There is one issue. On code line.  OpenCL.IsEnabled = true; it fails on macos. It says that dll q8-arm64 is not found. Which I understand, because mac uses it's in house chip based on arm64 architecture. But The application should work on multiple platforms including arm64 and x64 platforms. That's why I installed q8-AnyCPU in the hopes, that it would involve all of the architectures. But it seems like it's not working. From what I read in documentation it seems AnyCPU means either x86 or x64, but not arm64. Is there any workaround so I could use the same application on mac computers and Windows computers ? Worst case scenario I will just change out the nuget before deploying to mac, but it would be nice if there is some kind of solution besides this ?

**Comment by dlemstra** [comment]
There is support for `arm64` in the AnyCPU package. I have updated the installation docs to also mention `arm64`. I suspect there is an issue with your deployment and that is why the native code is not executed. Is the native library in your output folder?

**Reply by justasxz** [reply]
![image](https://github.com/user-attachments/assets/2f716564-4fe3-4847-96d2-6dc134b96ce0)
Here is the image of an error, and yes I can't find this Magick.Native-Q8-arm64.dll in the output folder, but I would believe it would be auto generated :/ Because other dlls are there automatically like any cpu I only saw Magick.Native-Q8.arm64.items (or similarly named file). Compiling trough VS Code, well maybe it's deployment. I could probably add the file manually, where do I get only that one file :?

**Reply by dlemstra** [reply]
The Native file should be in your output folder. You can force a copy with one of the options mentioned in the [cross platform](https://github.com/dlemstra/Magick.NET/blob/main/docs/CrossPlatform.md) docs.

**Reply by justasxz** [reply]
> The Native file should be in your output folder. You can force a copy with one of the options mentioned in the [cross platform](https://github.com/dlemstra/Magick.NET/blob/main/docs/CrossPlatform.md) docs.

Hi, sorry for late reply, after adding the line to copyNativeMacOS it worked initially, but after cleaning and rebuilding this is the error I am getting:

building for Mac Catalyst, but linking in dylib built for macOS, file 'obj/Debug/net8.0-maccatalyst/maccatalyst-arm64/nativelibraries/Contents/MonoBundle/Magick.Native-Q8-arm64.dll.dylib'

Any advice ?

**Reply by dlemstra** [reply]
It looks like you should target `net8.0-macos` instead of `net8.0-maccatalyst`?

**Reply by justasxz** [reply]
> It looks like you should target `net8.0-macos` instead of `net8.0-maccatalyst`?

It worked previously on macOS targeting macCatalyst it just decided one day to stop working without any real changes :/  :( And blazor hybrid (at least switching targets basically gave me hundred of errors and made project unsalvagable) only supports targeting maccatalyst. :/ I guess I am stuck thank you for your assistance, if you come up with something please let me know. Thanks in advance, I will try to somehow work around.
